lib/SPIRV/libSPIRV/SPIRVInstruction.h

```diff
@@ -2899,6 +2898,32 @@ _SPIRV_OP(UDotAccSatKHR, true, 6, true, 3)
 _SPIRV_OP(SUDotAccSatKHR, true, 6, true, 3)
 #undef _SPIRV_OP
 
+class SPIRVBitOp : public SPIRVInstTemplateBase {
+public:
+  SPIRVCapVec getRequiredCapability() const override {
+    if (Module->isAllowedToUseExtension(ExtensionID::SPV_KHR_bit_instructions))
+      return getVec(CapabilityBitInstructions);
+
+    return getVec(CapabilityShader);
```

Comment:
It's a bit strange to see `Shader` capability in the LLVM IR to SPIR-V translator. Is it a typo and if it's not for my curiosity could you explain it?

Reply:
I mean, I acknowledge that these instructions are added in SPIR-V under `shader` capability. But shouldn't the translator gracefully reject them in case if the extension isn't enabled instead of processing them anyway under `shader `capability as it normally doesn't work with the translator?

Reply:
> It's a bit strange to see Shader capability in the LLVM IR to SPIR-V translator. Is it a typo and if it's not for my curiosity could you explain it?

It is not a typo: the translator does have very limited support for some Shader / Matrix functionality.  See also the (long-standing) issue #30.

Returning `CapabilityShader` here preserves the existing behaviour of the translator, as evidenced by for example `test/transcoding/OpBitReverse_i32.ll`.

Reply:
Okay, thanks!